shlr/tcc/tccpp.c

```diff
@@ -2812,16 +2812,20 @@ static int macro_subst_tok(TokenString *tok_str,
 		goto add_cstr;
 	} else if (tok == TOK___DATE__ || tok == TOK___TIME__) {
 		time_t ti;
-		struct tm *tm;
+		struct tm tminfo;
 
 		time(&ti);
-		tm = localtime(&ti);
+#if __WINDOWS__
```

Comment:
Instead of this, use `cc.has_function` in `meson.build` to check if `localtime_r` is available and define a HAVE_LOCALTIME_R in rz_userconf.h. Do the same for `localtime_s` and choose what to do here based on whether there is HAVE_LOCALTIME_R, HAVE_LOCALTIME_S or none at all.

This way even if you are on windows but for some reason localtime_r is supported, everything still works.

Reply:
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/localtime-localtime32-localtime64?view=msvc-160

Windows doesn't have the same API. I don't know why but they choose localtime_s and linux choose localtime_r

Reply:
I know, but maybe mingw/msys2 or clang-cl might add that function somehow/somewhen(if they don't already), so it's better/safer to just check for the function at build-time, rather than relying on you/us knowing which platform supports what.

Reply:
The file is in shlr/*. It doesn't have access to librz/include/* def

Reply:
It seems that tcc.h is including rz_types.h, so I guess you can include rz_userconf.h anyway.

Reply:
I replace the macro with rz_locatime_r.

Reply:
Even better, yeah!